java-frontend/src/main/java/org/sonar/java/se/ExplodedGraphWalker.java

```diff
@@ -829,8 +829,7 @@ private void executeAssignement(AssignmentExpressionTree tree) {
     ProgramState.Pop unstack;
     SymbolicValue value;
 
-    boolean isSimpleAssignment = ExpressionUtils.isSimpleAssignment(tree) || ExpressionUtils.isSelectOnThisOrSuper(tree);
-    if (isSimpleAssignment) {
+    if (tree.is(Tree.Kind.ASSIGNMENT)) {
```

Comment:
Discussed mostly offline at whiteboard, just short summary for posterity

- `this.x` is not evaluated in EGWalker, because "simple assignment" CFG doesn't include evaluation of the LHS, see https://github.com/SonarSource/sonar-java/blob/5cd996433742016c3e4c467daaac8bd5c825e10c/java-frontend/src/main/java/org/sonar/java/cfg/CFG.java#L593-L593. 

- Indeed in case of member select, which is not using this, the number of unstacked values is incorrect.